week0003/thainguyentran/CSharp/week0003.cs

```diff
@@ -0,0 +1,66 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace DVRK
+{
+    class Program
+    {
+        public static Dictionary<string, double> clients = new Dictionary<string, double>();
+        const char ACTION_ADDCLIENT = 'A';
+        const char ACTION_DEPOSIT = 'D';
+        const char ACTION_WITHDRAW = 'W';
+        
+        static void Main(string[] args)
+        {
+            string input = null;
+            do {
+                input = Console.ReadLine();
+                Console.WriteLine(bankingProcess(input));
+            } while (input != null) ;
+        }
+        
+        public static string bankingProcess (string input)
+        {
+            string[] keyword = input.Split(' ');
+            char Trigger = Convert.ToChar(keyword[0]);
+            string Name = keyword[1];
+            double Amount = Convert.ToDouble(keyword[2]);
+            
+            switch (Trigger)
+            {
+                case ACTION_ADDCLIENT:
+                    if (Name.All(char.IsLetter) && Amount > 0)
```

Comment:
Code duplication. This line repeated 3 times. Also note that if amount is not positive then return `False`, not `Invalid`. `Invalid` only for bad actions.